examples/minidom/fake_codegen.rs

```diff
@@ -71,15 +74,15 @@ static WINDOW_CLASS: JSClass = JSClass {
         mayResolve: None,
         resolve: None,
         setProperty: None,
-        trace: Some(trace_jsobject_with_native_data),
+        trace: Some(JS_GlobalObjectTraceHook),
```

Comment:
OK, we're into jsapi weirdness here, why does this pick up josephine's object tracer? Does this assume that the global was created by josephine?

Reply:
The JSAPI requires the global object to use JS_GlobalObjectTraceHook, which does some important stuff and then delegates to the trace hook specified in the CompartmentOptions that were provided when the global was created. The new method I've added to JSInitializer to provide a global trace hook defaults to trace_jsobject_with_native_data, so it should still be invoked for the global object.

Reply:
Okay, that makes sense.